### PR TITLE
feat(cassandra): Refactor session configuration.

### PR DESCRIPTION
**Pull request by broneill**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :**
A session provider is used to customize how Cassandra sessions are created, but this requires that a provider be passed around and there's no common configuration strategy.

**New behavior :**
The session provider is no longer passed around, but instead the class is obtained from config. A session is created early and passed to anything that needs it.

**BREAKING CHANGES**
Any deployment which relies on a custom session provider needs to define the new "session-provider-fqcn" config setting under the "cassandra" section.

### CODE REVIEW

File: spark-jobs/src/main/scala/filodb/downsampler/DownsamplerSettings.scala [hunk 1]
```diff
@@ -25,8 +25,6 @@ object DownsamplerSettings extends StrictLogging {
   val downsamplerConfig = filodbConfig.getConfig("downsampler")
   logger.info(s"Loaded following downsampler config: ${downsamplerConfig.root().render()}" )
 
-  val sessionProvider = downsamplerConfig.as[Option[String]]("cass-session-provider-fqcn")
```
Comment (vishramachandran) [inline spark-jobs/src/main/scala/filodb/downsampler/DownsamplerSettings.scala]: Reminder ... need to remove this from our config files where provided

File: cassandra/src/main/scala/filodb.cassandra/CassandraTSStoreFactory.scala [hunk 2]
```diff
@@ -18,29 +18,18 @@ import filodb.core.store.{NullColumnStore, NullMetaStore}
...
-/**
-  * A StoreFactory for a TimeSeriesMemStore with Cassandra for metadata, but NullColumnStore for
-  * disabling write of chunks to a persistent column store. This can be used in environments
-  * where we would like to test in-memory aspects of the store and ignore persistence and
-  * on-demand-paging.
-  *
-  * @param config a Typesafe Config, not at the root but at the "filodb." level
-  * @param ioPool a Monix Scheduler, recommended to be the standard I/O pool, for scheduling asynchronous I/O
-  */
-class NonPersistentTSStoreFactory(config: Config, ioPool: Scheduler) extends StoreFactory {
```
[the remark this answers is not in the document]
Comment (broneill) [reply cassandra/src/main/scala/filodb.cassandra/CassandraTSStoreFactory.scala]: I added it back.

File: cassandra/src/test/scala/filodb.cassandra/columnstore/CassandraColumnStoreSpec.scala [hunk 3]
```diff
@@ -18,13 +18,15 @@ import filodb.memory.format.ZeroCopyUTF8String._
...
+import filodb.cassandra.DefaultFiloSessionProvider
 import filodb.cassandra.metastore.CassandraMetaStore
 
 class CassandraColumnStoreSpec extends ColumnStoreSpec {
   import NamesTestData._
 
-  lazy val colStore = new CassandraColumnStore(config, s)
-  lazy val metaStore = new CassandraMetaStore(config.getConfig("cassandra"))
+  lazy val session = new DefaultFiloSessionProvider(config.getConfig("cassandra")).session
+  lazy val colStore = new CassandraColumnStore(config, s, session)
```
Comment (velvia) [inline cassandra/src/test/scala/filodb.cassandra/columnstore/CassandraColumnStoreSpec.scala]: I think maybe this test should just use the `AllTablesTest` trait and avoid this duplication.

Comment (broneill) [reply cassandra/src/test/scala/filodb.cassandra/columnstore/CassandraColumnStoreSpec.scala]: This ends up creating a bunch of new compilation conflicts. Refactoring the tests is a bit more work and out of scope.

**Pull request merged**